src/Microsoft.Data/DataFrame.IO.cs

```diff
@@ -127,23 +127,28 @@ private static Type MaxKind(Type a, Type b)
             using (var st = createStream())
             {
                 string line = st.ReadLine();
-                int nbline = 0;
-                while (line != null && (numRows == -1 || rowline < numRows))
+                long nbline = 0;
+                while (line != null)
```

Comment:
You are now looping over all the rows twice, correct?

Reply:
I wonder if we could use a `ReadOnlySpan<char>` to parse the file, instead of creating a new string for every line.

---
In reply to: [321825779](https://github.com/dotnet/corefxlab/pull/2717#discussion_r321825779) [](ancestors = 321825779)

Reply:
> You are now looping over all the rows twice, correct?

Yup.

> I wonder if we could use a ReadOnlySpan<char> to parse the file, instead of creating a new string for every line.

Hmm, maybe? This ReadCsv method is just a method that I put together so we can work on real files on disk. At the moment, the `Split` calls on each line are taking up a sizeable chunk of time. Maybe I should file an issue to revamp this method now that we have decent capabilities in the DataFrame

Reply:
> At the moment, the Split calls on each line are taking up a sizeable chunk of time.

Right, this is because new `string` instances are being created, and their contents are being copied out, for each section in the file. Using a span to parse will make it faster because you don't need to create all those intermediate `string` instances.

Reply:
Filed https://github.com/dotnet/corefxlab/issues/2720